File: planik-android/app/src/main/java/pl/planik/local/mapper/to/model/UserEntityToModelMapper.kt
package pl.planik.local.mapper.to.model

import pl.planik.domain.model.User
import pl.planik.local.entity.UserEntity
import pl.planik.local.mapper.Mapper
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class UserEntityToModelMapper @Inject constructor() : Mapper<UserEntity, User> {
  override fun map(value: UserEntity): User = User(
    id = value.id,
    uuid = value.uuid,
    nickname = value.nickname,
    createdAt = value.createdAt,
    updatedAt = value.updatedAt
  )
}

File: planik-flutter/planik/lib/local/mapper/user_entity_to_model_mapper.dart
import 'package:planik/domain/model/user.dart';
import 'package:planik/local/entity/user_entity.dart';
import 'package:planik/local/mapper/mapper.dart';

class UserEntityToModelMapper implements Mapper<UserEntity, User> {
  @override
  User map(UserEntity entity) {
    return User(
      id: entity.id!,
      uid: entity.uid,
      nickname: entity.nickname,
      createdAt: DateTime.parse(entity.createdAt),
      updatedAt: entity.updatedAt != null ? DateTime.parse(entity.updatedAt!) : null,
    );
  }
}
